Zend/zend_builtin_functions.c

```diff
@@ -933,6 +933,12 @@ ZEND_FUNCTION(method_exists)
 			}
 			RETURN_TRUE;
 		}
+	} else {
+	    /* Returns true for fake Closure::__invoke */
+	    if (ce == zend_ce_closure
+	        && zend_string_equals_literal(method_name, ZEND_INVOKE_FUNC_NAME)) {
```

Comment:
This is also wrong in the code above, but the comparison should be against `lcname`. `CLOSURE::__INVOKE` should work (sigh).

Reply:
Or alternatively it should use `_ci` comparison, if we don't want to extend lifetime of lcname.

Reply:
Ah, fixed ... and included those cases in the test for this bug.